kernel/appenv.c

```diff
@@ -0,0 +1,252 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <myst/args.h>
+#include <myst/eraise.h>
+#include <myst/file.h>
+#include <myst/fsgs.h>
+#include <myst/json.h>
+#include <myst/kernel.h>
+#include <myst/printf.h>
+
+#define APPENV_FILENAME "appenv.json"
+
+typedef struct json_callback_data
+{
+    myst_args_t env;
+} json_callback_data_t;
+
+static int _concat_env_variables(
+    const char* value,
+    myst_args_t* new_envp,
+    int pos)
+{
+    char* tuple;
+    const char* curr = new_envp->data[pos];
+
+    if (!curr)
+        return -EINVAL;
+
+    /* Append the APPENV_FILENAME value of the var to the config.json value */
+    asprintf(&tuple, "%s:%s", new_envp->data[pos], value);
+
+    new_envp->data[pos] = tuple;
+
+    free((char*)curr);
+    return 0;
+}
+
+static void _trace(
+    json_parser_t* parser,
+    const char* file,
+    unsigned int line,
+    const char* func,
+    const char* message)
+{
+    (void)parser;
+    printf("%s(%u): %s(): %s\n", file, line, func, message);
+}
+
+static json_result_t _json_read_callback(
+    json_parser_t* parser,
+    json_reason_t reason,
+    json_type_t type,
+    const json_union_t* un,
+    void* callback_data)
+{
+    json_result_t result = JSON_UNEXPECTED;
+    json_callback_data_t* cbdata = (json_callback_data_t*)callback_data;
+
+    switch (reason)
+    {
+        case JSON_REASON_NONE:
+        {
+            result = JSON_UNEXPECTED;
+            goto done;
+        }
+        case JSON_REASON_NAME:
+        {
+            break;
+        }
+        case JSON_REASON_BEGIN_OBJECT:
+        {
+            break;
+        }
+        case JSON_REASON_END_OBJECT:
+        {
+            break;
+        }
+        case JSON_REASON_BEGIN_ARRAY:
+        {
+            break;
+        }
+        case JSON_REASON_END_ARRAY:
+        {
+            break;
+        }
+        case JSON_REASON_VALUE:
+        {
+            if (json_match(parser, "Env") == JSON_OK)
+            {
+                if (type == JSON_TYPE_STRING)
+                {
+                    if (myst_args_append1(&cbdata->env, un->string) != 0)
+                    {
+                        result = JSON_OUT_OF_MEMORY;
+                        goto done;
+                    }
+                }
+                else
+                {
+                    result = JSON_BAD_SYNTAX;
+                    goto done;
+                }
+            }
+            else
+            {
+                result = JSON_OK;
+                goto done;
+            }
+
+            break;
+        }
+    }
+
+    result = JSON_OK;
+
+done:
+    return result;
+}
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wstack-usage="
```

Comment:
Why the stack size warning has to be turned off? Any huge stack variable should be allocated on the heap using the `locals` structure. 

This function is only called once during the entry of kernel so maybe it's fine to keep the variable on the stack. But I am curious which variable is costing the stack.

Reply:
Was able to turn this off, thank you for your help with the suggestions